scrapers_next/pa/committees.py

```diff
@@ -0,0 +1,171 @@
+from spatula import CSS, HtmlPage, HtmlListPage, XPath, SelectorError
+from openstates.models import ScrapeCommittee
+
+
+class HouseCommitteeDetail(HtmlPage):
+    example_source = "https://www.legis.state.pa.us/cfdocs/CteeInfo/index.cfm?Code=32&CteeBody=H&SessYear=2021"
+    example_name = "Aging & Older Adult Services"
+    example_input = ScrapeCommittee(
+        name=example_name, classification="committee", chamber="lower"
+    )
+
+    def process_page(self):
+        com = self.input
+        try:
+            # This section has the chair memebers the regular, democratic and minority and the roles
+            # main chair
+            Chair_Member = (
+                CSS("div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText a")
+                .match(self.root)[0]
+                .text.strip()
+            )
+            # main chair role
+            Chair_Member_rolez = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[0]
+                .text.strip()
+            )
+            com.add_member(Chair_Member, Chair_Member_rolez)
+            # Democratic Chair member and or the minority chair member
+            Demo_Chair_Member = (
+                CSS("div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText a")
+                .match(self.root)[1]
+                .text.strip()
+            )
+            # Democratic Chair member and or the minority chair member role
+            Demo_Chair_Member_rolez = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[1]
+                .text.strip()
+            )
+            com.add_member(Demo_Chair_Member, Demo_Chair_Member_rolez)
+        except IndexError:
+            pass
+        # Regular majority members and their roles
+        Majority_Members = XPath("//div[12]/div/div/div/div[2]").match(self.root)
```

Comment:
We talked about how there is a CSS selector that looks like it would be more appropriate to use to find the majority members, as well as another CSS selector for the minority members. Did you try that?

Reply:
The CSS selector (CSS("div.Widget CteeInfo-MajortityList")) didn't work it through back an error but as I started trying more selectors the only CSS selectors that worked were (CSS("div:nth-child(13) div div div div:nth-child(2)") and ("div a") for name and ("div.position") for the role/position. This one is much to similar to the XPath so i just left the XPath there.

Reply:
OK, so when you say "it threw back an error" that's an example of a lack of specifics - when I asked for "a series of commands/code you tried from a spatula shell session (with URL specified)" I'm looking for something like this:

```
(openstates-scrapers) jesse@codepc:~/repo/civiceagle/openstates$ spatula shell "https://www.legis.state.pa.us/cfdocs/CteeInfo/index.cfm?Code=32&CteeBody=H&SessYear=2021"
spatula 0.8.7 shell
available selectors: CSS, SimilarLink, XPath
local variables
---------------
url: https://www.legis.state.pa.us/cfdocs/CteeInfo/index.cfm?Code=32&CteeBody=H&SessYear=2021
resp: requests Response instance
root: `lxml HTML element` <html>
Python 3.9.5 (default, Jun  8 2021, 14:08:29) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.30.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: maj_list = CSS(".Widget.CteeInfo-MajorityList").match_one(root)
---------------------------------------------------------------------------
SelectorError                             Traceback (most recent call last)
~/.pyenv/versions/3.9.5/envs/openstates-scrapers/lib/python3.9/site-packages/spatula/cli.py in <module>
----> 1 maj_list = CSS(".Widget.CteeInfo-MajorityList").match_one(root)

~/.pyenv/versions/3.9.5/envs/openstates-scrapers/lib/python3.9/site-packages/spatula/selectors.py in match_one(self, element)
     78         :param element: Element to search within.
     79         """
---> 80         return self.match(element, num_items=1)[0]
     81 
     82     def get_items(self, element: _Element) -> Iterator[_Element]:  # pragma: no cover

~/.pyenv/versions/3.9.5/envs/openstates-scrapers/lib/python3.9/site-packages/spatula/selectors.py in match(self, element, min_items, max_items, num_items)
     55 
     56         if num_items is not None and len(items) != num_items:
---> 57             raise SelectorError(
     58                 f"{self} on {_display(element)} got {len(items)} results, "
     59                 f"expected {num_items}"

SelectorError: CSS(.Widget.CteeInfo-MajorityList) on <html lang='en'> @ line 2 got 2 results, expected 1
```

Generally when working with other developers it is better to provide too much detail than not enough. Plus, the task of documenting your own problem in detail often helps in your own troubleshooting process (see for example [rubber duck debugging](https://en.wikipedia.org/wiki/Rubber_duck_debugging).

The CSS selector you have above contains two errors: 
- `CteeInfo-MajortityList` is misspelled: the class from the site is `.CteeInfo-MajorityList` Please be sure to copy-paste strings from the website rather than retype, so you always get exactly the same string
- A space between selectors indicates that you are selecting a *child* of the first selector. So `div.Widget .CteeInfo-MajorityList` will look for a child element of `div.Widget` that has the class `.CteeInfo-MajorityList`. In this case what you want is an element that has both classes .Widget and .CteeInfo-MajorityList, so instead it should be `div.Widget.CteeInfo-MajorityList` (no space in between - they are chained together).

There is a working solution here using CSS selectors. Please try again :)